server/src/main/java/org/opensearch/search/aggregations/bucket/histogram/DateHistogramAggregator.java

```diff
@@ -171,6 +180,51 @@ public LeafBucketCollector getLeafCollector(LeafReaderContext ctx, LeafBucketCol
         if (optimized) throw new CollectionTerminatedException();
 
         SortedNumericDocValues values = valuesSource.longValues(ctx);
+        CompositeIndexFieldInfo supportedStarTree = getSupportedStarTree(this.context);
+        if (supportedStarTree != null) {
+            StarTreeValues starTreeValues = getStarTreeValues(ctx, supportedStarTree);
+            assert starTreeValues != null;
+
+            FixedBitSet matchingDocsBitSet = StarTreeFilter.getPredicateValueToFixedBitSetMap(starTreeValues, "@timestamp_month");
+
+            SortedNumericStarTreeValuesIterator valuesIterator = (SortedNumericStarTreeValuesIterator) starTreeValues
+                .getDimensionValuesIterator("@timestamp_month");
+
+            SortedNumericStarTreeValuesIterator metricValuesIterator = (SortedNumericStarTreeValuesIterator) starTreeValues
+                .getMetricValuesIterator("startree1__doc_count_doc_count_metric");
+
+            int numBits = matchingDocsBitSet.length();
+
+            if (numBits > 0) {
+                for (int bit = matchingDocsBitSet.nextSetBit(0); bit != DocIdSetIterator.NO_MORE_DOCS; bit = (bit + 1 < numBits)
+                    ? matchingDocsBitSet.nextSetBit(bit + 1)
+                    : DocIdSetIterator.NO_MORE_DOCS) {
+
+                    if (!valuesIterator.advanceExact(bit)) {
+                        continue;
+                    }
+
+                    for (int i = 0, count = valuesIterator.entryValueCount(); i < count; i++) {
+                        long dimensionValue = valuesIterator.nextValue();
+
+                        if (metricValuesIterator.advanceExact(bit)) {
+                            long metricValue = metricValuesIterator.nextValue();
+
+                            long bucketOrd = bucketOrds.add(0, dimensionValue);
+                            if (bucketOrd < 0) {
+                                bucketOrd = -1 - bucketOrd;
+                                collectStarTreeBucket((StarTreeBucketCollector) sub, metricValue, bucketOrd, bit);
```

Comment:
Do we need to use `collectExistingStarBucket` / `collectStarBucket` paradigm here ?

Reply:
Can refactor to it if required. Both the utilities differ by `grow(bucketOrd + 1);` which I had used manually for now during POC but will abstract it in a better way.